backend-lib/src/main/rust/utils.rs

```diff
@@ -33,14 +33,30 @@ pub fn catch_unwind<F: FnOnce(&mut JNIEnv) -> R + UnwindSafe, R>(
     })
 }
 
-pub(crate) fn java_string_to_rust(env: &mut JNIEnv, jstring: &JString) -> String {
-    env.get_string(jstring)
-        .expect("Couldn't get Java string!")
-        .into()
+pub(crate) fn java_bytes_to_rust(env: &JNIEnv, jbytes: &JByteArray) -> anyhow::Result<Vec<u8>> {
```

Comment:
This could be `jbytes: JByteArray` since it is effectively already a pointer, as above.

Reply:
Leaving as-is because the method it wraps takes a reference, and this method doesn't do anything special on top (the type-specific helper methods do by comparison; they effectively are converting type, so we want to move there).